emacs-25.1 fails on os x

```
[0K$ make -f emacs-travis.mk install_emacs install_cask

Download Emacs 25.1 from https://ftp.gnu.org/gnu/emacs/emacs-25.1.tar.xz

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed


  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  6 40.5M    6 2832k    0     0  3148k      0  0:00:13 --:--:--  0:00:13 3146k
 24 40.5M   24  9.9M    0     0  5349k      0  0:00:07  0:00:01  0:00:06 5347k
 46 40.5M   46 18.9M    0     0  6661k      0  0:00:06  0:00:02  0:00:04 6659k
 70 40.5M   70 28.6M    0     0  7535k      0  0:00:05  0:00:03  0:00:02 7534k
 97 40.5M   97 39.3M    0     0  8222k      0  0:00:05  0:00:04  0:00:01 8221k
100 40.5M  100 40.5M    0     0  8283k      0  0:00:05  0:00:05 --:--:-- 9408k

Install Emacs 25.1

configure: error: Non-toolkit scroll bars are not implemented for Nextstep.

make: *** [install_emacs] Error 1



travis_time:end:1121f4f8:start=1474373164000000000,finish=1474373189000000000,duration=25000000000
[0K

[31;1mThe command "make -f emacs-travis.mk install_emacs install_cask" failed and exited with 2 during .[0m
```

--- a/emacs-travis.mk
+++ b/emacs-travis.mk
@@ -50,6 +50,11 @@ EMACSCONFFLAGS += --with-modules
 endif
 endif
 
+ifeq ($(TRAVIS_OS_NAME),osx)
+# The "games" group does not exist on macOS, so we configure --with-gameuser to the current user (i.e., "travis")
+EMACSCONFFLAGS += --with-gameuser=$(USER)
+endif
+
 # Clone Emacs from the Github mirror because it's way faster than upstream
 EMACS_GIT_URL = https://github.com/emacs-mirror/emacs.git
 # Emacs FTP URL.  Prereleases are on alpha.gnu.org
@@ -96,6 +101,10 @@ export REPORTED_EMACS_VERSION
 .PHONY: test
 
 install_gnutls:
+ifeq ($(TRAVIS_OS_NAME),osx)
+#	GnuTLS can be upgraded with Homebrew instead of apt-get (which obviously does not exist on macOS)
+	brew upgrade gnutls
+else
 	@echo "Install GnuTLS 3"
 	@sudo apt-get -qq update
 	@sudo apt-get install -y build-essential nettle-dev libgmp-dev
@@ -106,6 +115,7 @@ install_gnutls:
 	  && make -j$(MAKE_JOBS) $(SILENT) \
 	  && sudo make install $(SILENT) \
 	  && sudo ln -s /usr/local/lib/libgnutls.so.28 /usr/lib/libgnutls.so.28
+endif
 
 download_emacs_stable:
 	@echo "Download Emacs $(EMACS_VERSION) from $(EMACS_TAR_URL)"
@@ -135,6 +145,10 @@ endif
 install_emacs:
 	@echo "Install Emacs $(EMACS_VERSION)"
 	@make -j$(MAKE_JOBS) -C "$(EMACS_DIR)" V=0 install $(SILENT)
+ifeq ($(TRAVIS_OS_NAME),osx)
+#	To pretend that an up-to-date emacs exists in $HOME/bin, we must link it out of Emacs.app
+	ln -s $(HOME)/emacs/$(EMACS_VERSION)/nextstep/Emacs.app/Contents/MacOS/Emacs $(HOME)/bin/emacs
+endif
 
 # Run configure (and download) only if directory is absent
 ifeq ($(wildcard $(EMACS_DIR)/.),)

--- a/emacs-travis.mk
+++ b/emacs-travis.mk
@@ -24,9 +24,6 @@ TEXINFO_VERSION ?= 6.1
 EMACS_VERSION ?= 25.3
 VERBOSE ?= no
 MAKE_JOBS ?= 2
-# Build a minimal Emacs with no special flags, to build as fast as possible
-EMACSCONFFLAGS ?= --with-x-toolkit=no --without-x --without-all --with-xml2 \
-	CFLAGS='-O2 -march=native' CXXFLAGS='-O2 -march=native'
 
 ifeq ($(VERBOSE),yes)
 SILENT=
@@ -41,6 +38,17 @@ PRE_RELEASE_PART := $(word 2,$(VERSION_PARTS))
 MAJOR_VERSION := $(word 1,$(subst ., ,$(EMACS_VERSION)))
 # Whether the version is a release candidate
 PRETEST ?= $(findstring rc,$(PRE_RELEASE_PART))
+
+# Build a minimal Emacs with no special flags, to build as fast as possible
+ifndef EMACSCONFFLAGS
+EMACSCONFFLAGS := --with-x-toolkit=no --without-x --without-all --with-xml2 \
+	CFLAGS='-O2 -march=native' \
+	CXXFLAGS='-O2 -march=native'
+
+ifeq ($(shell test ( $(EMACS_VERSION) -eq snapshot ) -o ( $(MAJOR_VERSION) -ge 25 ); echo $$?),0)
+EMACSCONFFLAGS += --with-modules
+endif
+endif
 
 # Clone Emacs from the Github mirror because it's way faster than upstream
 EMACS_GIT_URL = https://github.com/emacs-mirror/emacs.git

